自動更新機能の実装

shukan0728 opened the pull request:
WHAT
自動更新機能の実装
WHY
チャット機能改善のため

shukan0728 committed:
自動更新の実装

shukan0728 committed:
自動更新の実装

shukan0728 commented:
https://gyazo.com/592158c9597164bb73bdb47c8b31b612

MrUrakou commented on app/assets/javascripts/message_auto.js:
```diff
@@ -0,0 +1,61 @@
+
+  $(function(){
+    $(function(){
+      if (location.href.match(/\/groups\/\d+\/messages/)){
```
pathnameですね。hrefだとデプロイしてパスが変わると動かないので。

MrUrakou commented on app/views/messages/index.json.jbuilder:
```diff
@@ -0,0 +1,8 @@
+json.user_name message.user.name
+json.content message.content
+json.created_at message.created_at
+json.id message.image
```
インデント修正しましょう。

MrUrakou commented on app/views/messages/index.json.jbuilder:
```diff
@@ -0,0 +1,8 @@
+json.created_at message.created_at
+json.id message.image
+end
+
```
空行は末尾の改行1行だけにしましょう。

MrUrakou commented on config/initializers/time_formats.rb:
```diff
@@ -0,0 +1,6 @@
+
```
こちらの空行も不要です。

MrUrakou commented on app/controllers/messages_controller.rb:
```diff
@@ -5,6 +5,10 @@ class MessagesController < ApplicationController
     @messages = @group.messages.includes(:user)
+    respond_to do |format|
+      format.html
+      format.json { @add_messages = @messages.where('id > ?',params[:id]) }
```
,のあとは半角スペースです。

MrUrakou commented on app/assets/javascripts/user_inc.js:
```diff
@@ -56,6 +56,7 @@ $(function() {
     var id = $(this).data('user-id');
     var name = $(this).data('user-name');
     var insertHTML = buildHTML(id, name);
+    console.log(name);
```
こちら消しましょう。

shukan0728 committed:
＃１修正/自動更新機能の実装

・不要なインデント・空行の修正
・スクロールの挙動がおかしかったため、修正
→ setIntervalを定義、clearIntervalの追加
・href→pathname

shukan0728 commented:
https://gyazo.com/12c71fbd5c67911f62cab2b6457c7106

MrUrakou commented on app/views/messages/index.json.jbuilder:
```diff
@@ -0,0 +1,8 @@
+  json.created_at message.created_at
+  json.id message.image
+end
+
```
この空行は削除して、末尾の改行1行にしましょう。

MrUrakou commented on app/controllers/groups_controller.rb:
```diff
@@ -10,7 +11,9 @@ def new
 end
 
 def create
+  binding.pry
```
デバック消しましょう。

MrUrakou commented on app/controllers/groups_controller.rb:
```diff
@@ -23,6 +26,7 @@ def edit
 end
 
 def update
+  # binding.pry
```
こちらも消してください。

MrUrakou commented on app/assets/javascripts/message_auto.js:
```diff
@@ -0,0 +1,62 @@
+      })
+      .done(function(data){
+        if (data.length){
+          $.each(data, function(i, data){
```
forEach使いましょう。
iは無いと動かないですか？
```
.done(function(data){
 data.forEach(function(element){
  var html = buildHTML(element);
```
みたいな記述で動きませんか？

shukan0728 committed:
#2修正/自動更新機能の実装

shukan0728 commented:
https://gyazo.com/ffb0357dab5c44f5be53e3fab45130e4

MrUrakou commented on app/views/messages/index.json.jbuilder:
```diff
@@ -0,0 +1,7 @@
+  json.user_name message.user.name
+  json.content message.content
+  json.created_at message.created_at
+  json.id message.image
```
ここmessage.image.urlです。
渡すのはあくまでurlです。

MrUrakou commented on app/controllers/groups_controller.rb:
```diff
@@ -32,7 +34,7 @@ def update
 private
   def group_params
-  params.require(:group).permit(:name, { :user_ids => [] })
+  params.require(:group).permit(:name, {user_ids: []})
```
インデント深くしましょう。

MrUrakou commented on app/assets/javascripts/message_auto.js:
```diff
@@ -0,0 +1,62 @@
+          return html;
+    }
+    function scroll() {
+      $('.chat-contents').stop().animate({scrollTop: $('.chat-contents')[0].scrollHeight});
```
stop()が必要な理由って何ですか？

shukan0728 committed:
#3修正/自動更新機能の実装
